NeoML/src/TraditionalML/KMeansClustering.cpp

```diff
@@ -57,6 +57,171 @@ static CPtr<CDnnBlob> createWeightBlob( IMathEngine& mathEngine, const IClusteri
 
 //-------------------------------------------------------------------------------------------------------------
 
+template <int Size>
```

Comment:
What's the reason for template argument? In both cases it will work effectively if it will be greater or equal to the `threadCount`. You initialiize it with different values in different places? Why not just `CFastArray<double, 16> localInertia{}`?

Reply:
Same question about `CFastArray` - why `Size` is template parameter? It can be runtime without penalty, IMHO.

Reply:
fixed